importlib_metadata/__init__.py

```diff
@@ -153,6 +155,43 @@ def __reduce__(self):
         )
 
 
+class EntryPoints(tuple):
```

Comment:
the pickle and json shape of both of these objects changes in a breaking way

additionally, I usually find it's best to [set `__slots__ = ()` when extending `tuple` to preserve immutability](https://youtu.be/BSNd_kxHXL8)

Reply:
Pickle and JSON shape of entry point containers isn't a tested or documented interface. Do you think it should be? My instinct is it's YAGNI or at least an unsupported expectation.

Good suggestion on `__slots__`. That seems like a good idea. Usually, I prefer a minimal implementation, but in this case, it seems reasonable.

Reply:
it isn't tested or documented, but since it is part of the standard library you'll want to adhere to best practices for cpython which essentially boils down to "if it could be json'd or pickled, people are already doing that and changing that is a breaking change" -- or at least that has been my experience working with contributions to cpython

Reply:
In relation to CPython, this change will go into 3.10, and pickle compatibility and shape is not guaranteed across minor releases. Moreover, I'm struggling how to think to improve the situation without breaking that expectation. Any introduction of a new class will automatically be incompatible on prior versions (unable to deserialize there). I appreciate the feedback, but I think this is a necessary breakage.